### PR TITLE
Nd to complete 6 & 7 then finished

### PR DESCRIPTION
**Pull request by LaFrish**

### CODE REVIEW

**Comment by LaFrish** [comment]
complete 4 
comprehension 4.5

File: README.md [hunk 1]
```diff
@@ -25,7 +25,15 @@ Demonstrate calling the method with an argument of "young prince".
 
 Write your code here:
 ```ruby
-# code here
+Def offerRose(person)
```
Comment (jshawl) [inline README.md:28]: lowercase `def`

File: README.md [hunk 2]
```diff
@@ -25,7 +25,15 @@ Demonstrate calling the method with an argument of "young prince".
...
+puts "Would you take this rose and help out
+an old beggar, #{person}"
```
Comment (jshawl) [inline README.md:30]: move to same line

File: README.md [hunk 3]
```diff
@@ -25,7 +25,15 @@ Demonstrate calling the method with an argument of "young prince".
...
+end
+
+
+offerRose(young prince)
```
Comment (jshawl) [inline README.md:34]: just this one ^^

Comment (jshawl) [inline README.md:34]: young prince should be in quotes

File: README.md [hunk 4]
```diff
@@ -48,7 +56,9 @@ add her to the list of guests in the castle.
 
 Write your code here:
 ```ruby
-# code here
+town.residents.delete_at(1)
+castle.residents.push("Belle")
```
Comment (jshawl) [inline README.md:60]: shoudl be `town[:castle][:residents]`

File: README.md [hunk 5]
```diff
@@ -70,7 +80,9 @@ Belle is friends with Mrs. Potts
 
 Write your code here:
 ```ruby
-# code here
+friends.each do |friend|
+  puts "Belle is friends with" [:friend]
+end
```
Comment (jshawl) [inline README.md:85]: instead -  "Belle is friends with #{friend}"

File: README.md [hunk 6]
```diff
@@ -81,8 +93,13 @@ Describe the differences between unit and functional testing. What type of testi
 
 Your answer:
 ```text
+Unit testing check the small blocks of code to make sure that input get the specific results you want. With unit testing you check the design, architecture, maintainability and fewer bugs. The tests are individual methods are not intended to be longer than 5 lines.  Unit tests are run before functional tests.
+
+Functional tests look at the bigger picture and have a wider focus. These tests are for Functionalities on sites such as the specific view based on permission level, making sure forms and sign ins work.
+
+RSpec is a unit test that is a testing framework for Ruby. It is supposed to makes writing tests much more simple for developers. You write the test before you write your actual code to make sure you get the specific results that you want.
```
Comment (jshawl) [inline README.md:100]: :+1:

File: README.md [hunk 7]
```diff
@@ -112,8 +129,7 @@ end
 
 Your answer:
 ```text
-
-Replace this with your answer
+context and describe are literally the same thing. The reason to use both is to make the test more readable. Both describe and context are testing to say there was rooms/beds available for tenants.
```
Comment (jshawl) [inline README.md:132]: :+1:

File: README.md [hunk 8]
```diff
@@ -131,7 +147,24 @@ entities (no need to draw an ERD):
...
+                  |        |      
+               friends   posseses
+                with        &
+                PET       rubs     
+                          LAMP  --------> GENIE
+                                      Lives in a lamp & grants 3 wishes to person who rubs it's lamp
+
+
+Genie lives in lamp
+grants wishes to person
```
Comment (jshawl) [inline README.md:167]: :+1:

File: README.md [hunk 9]
```diff
@@ -142,7 +175,34 @@ SQL database. If you need an example, you can use: people and wishes
 
 Your answer:
 ```
-Replace this with your answer
+A schema is used in databases to show the databases information. It gives information such as each tables column name their data type and any constraints, or rules, for the columns.
```
Comment (jshawl) [inline README.md:178]: or structure :+1:

File: README.md [hunk 10]
```diff
@@ -167,5 +227,31 @@ Write code to do the following:
 
 Write your code here:
 ```ruby
-# code here
+1.
+genie = Genie.create(name: "Genie")
+genie_lamp = Lamp.create(wishes_remaining: "3")
```
Comment (jshawl) [inline README.md:232]: should associate lamp with genie `Genie.create(name:"", lamp: lamp)`

**Comment by jshawl** [comment]
:+1:

**Pull request closed without merging**